src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Component,
+  Host,
+  h,
+  Listen,
+  Prop,
+  Element,
+  Watch,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-radio-button",
+  styleUrl: "calcite-radio-button.scss",
+  shadow: true,
+})
+export class CalciteRadioButton {
+  constructor() {
+    this.check = this.check.bind(this);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties and Validators/Watchers
+  //
+  //--------------------------------------------------------------------------
+
+  /** The checked state of the radio button. */
+  @Prop({ mutable: true, reflect: true }) checked: boolean = false;
+  @Watch("checked")
+  onCheckedChange(newChecked: boolean, oldChecked: boolean) {
+    if (newChecked === true && oldChecked === false) {
+      this.uncheckOtherRadioButtonsInGroup();
+    }
+    if (newChecked !== this.input.checked) {
+      this.input.checked = newChecked;
+    }
+    this.calciteRadioButtonChange.emit();
+  }
+
+  /** The disabled state of the radio button. */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+  @Watch("disabled")
+  onDisabledChange(disabled: boolean) {
+    this.input.disabled = disabled;
+  }
+
+  /** The focused state of the radio button. */
+  @Prop({ mutable: true, reflect: true }) focused: boolean = false;
+  @Watch("focused")
+  onFocusedChange(focused: boolean) {
+    if (focused) {
+      this.input.focus();
+    } else {
+      this.input.blur();
+    }
+  }
+
+  /** The id attribute of the radio button.  When omitted, a globally unique identifier is used. */
+  @Prop({ reflect: true }) guid: string =
+    this.el.id || `calcite-radio-button-${guid()}`;
+
+  /** The name of the radio button.  <code>name</code> is passed as a property automatically from <code><calcite-radio-button-group></code>. */
+  @Prop({ reflect: true }) name!: string;
+  @Watch("name")
+  onNameChange(newName: string) {
+    if (newName !== this.input.name) {
```

Comment:
Why not always assign it, similar to other props?

Reply:
Great point.  You are right, the name will always be new so it should always assign it to the input without the check.  I'll update that.